src/services/db/GitFileSystemService.ts

```diff
@@ -400,6 +402,49 @@ export default class GitFileSystemService {
     })
   }
 
+  getFileExtension(fileName: string): string {
+    const parts = fileName.split(".")
+    if (parts.length > 1) {
+      return parts[parts.length - 1]
+    }
+    return "" // No extension found
+  }
+
+  getMimeType(fileExtension: string): string {
+    switch (fileExtension) {
+      case "svg":
+        return "image/svg+xml"
+      case "ico":
+        return "image/vnd.microsoft.icon"
+      default:
+        return `image/${fileExtension}`
+    }
+  }
+
+  readMediaFile(
+    siteName: string,
+    directoryName: string,
+    fileName: string
+  ): ResultAsync<MediaFileOutput, GitFileSystemError | NotFoundError> {
+    return this.read(
+      siteName,
+      `${directoryName}/${fileName}`,
+      "base64"
+    ).andThen((file) => {
```

Comment:
`file` implicitly has an  `any` type, we can specify `GitFile` as the type:
`andThen((file: GitFile)`